lasagne/layers/recurrent.py

```diff
@@ -107,7 +107,11 @@ class CustomRecurrentLayer(MergeLayer):
         :class:`lasagne.layers.Layer` instance which connects input to the
         hidden state (:math:`f_i`).  This layer may be connected to a chain of
         layers, which must end in a :class:`lasagne.layers.InputLayer` with the
-        same input shape as `incoming`.
+        same input shape as incoming, except for the first dimension: When
+        ``precompute_input == True`` (the default), it must be
+        ``incoming.output_shape[0]*incoming.output_shape[1]`` or `None`; when
+        ``precompute_input == False``, it must be ``incoming.output_shape[0]``
+        or `None`.
```

Comment:
> There are just two markup errors: "incoming" should have single backticks 

Oops, fixed.

> "None" should have double backticks because it's Python code. We've got markup errors like these all over the place, but let's get this right while we're at it!

Hm, then there are a few other places where that needs to be fixed too:
https://github.com/Lasagne/Lasagne/blob/master/lasagne/layers/recurrent.py#L150
https://github.com/Lasagne/Lasagne/blob/master/lasagne/layers/recurrent.py#L525
and in other files
https://github.com/Lasagne/Lasagne/blob/b22500e736bf1b8f773a5e11257c272880aa6dd6/lasagne/layers/input.py#L25
https://github.com/Lasagne/Lasagne/blob/b22500e736bf1b8f773a5e11257c272880aa6dd6/lasagne/layers/input.py#L28
https://github.com/Lasagne/Lasagne/blob/b22500e736bf1b8f773a5e11257c272880aa6dd6/lasagne/layers/input.py#L43
https://github.com/Lasagne/Lasagne/blob/7e39d52075772ef9351f9d9face40c1a0a955491/lasagne/layers/pool.py#L43
etc.

Perhaps we need to do a docstring audit?  Which convention would you prefer I use in this PR?